src/main/java/ch/njol/skript/expressions/ExprCoordinate.java

```diff
@@ -75,48 +80,57 @@ public Class<? extends Number> getReturnType() {
 	
 	@Override
 	@Nullable
-	public Class<?>[] acceptChange(final ChangeMode mode) {
-		if ((mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE) && getExpr().isSingle() && ChangerUtils.acceptsChange(getExpr(), ChangeMode.SET, Location.class))
-			return new Class[] {Number.class};
-		return null;
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if ((mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE) && getExpr().isSingle()) {
+			if (ChangerUtils.acceptsChange(getExpr(), ChangeMode.SET, Location.class))
+				return new Class[] {Number.class};
+		}
+		if(getExpr().getReturnType().equals(Chunk.class))
+			Skript.error("Can't set X/Z coordinates of chunks");
+		return new Class[0];
 	}
 	
 	@Override
-	public void change(final Event e, final @Nullable Object[] delta, final ChangeMode mode) throws UnsupportedOperationException {
-		assert delta != null;
-		final Location l = getExpr().getSingle(e);
-		if (l == null)
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) throws UnsupportedOperationException {
+		Object obj = getExpr().getSingle(event);
+		if (obj == null)
 			return;
-		double n = ((Number) delta[0]).doubleValue();
+		assert delta != null;
```

Comment:
After this you don't need to check obj instance in each switch case
```suggestion

		if (!(obj instanceof Location))
			return;
			
		assert delta != null;
```